core/state_transition.go

```diff
@@ -263,10 +263,17 @@ func (st *stateTransition) buyGas() error {
 	mgval := new(big.Int).SetUint64(st.msg.GasLimit)
 	mgval.Mul(mgval, st.msg.GasPrice)
 	var l1Cost *big.Int
-	if st.evm.Context.L1CostFunc != nil && !st.msg.SkipNonceChecks && !st.msg.SkipFromEOACheck {
-		l1Cost = st.evm.Context.L1CostFunc(st.msg.RollupCostData, st.evm.Context.Time)
-		if l1Cost != nil {
-			mgval = mgval.Add(mgval, l1Cost)
+	var operatorCost *uint256.Int
```

Comment:
I opened a PR merging into to the monorepo PR adds test coverage for this one, with a change that adds coverage for a balance underflow bug present in this logic.

https://github.com/yuwen01/optimism/pull/6

Reply:
I merged